### PR TITLE
include CWS functional tests in slack messages even if allowed to fail

### PR DESCRIPTION
**Pull request by paulcacheux**
### What does this PR do?

We would like to include the CWS functional tests failure in the Gitlab slack messages even for those that are allowed to fail. This PR adds a new check when including failing jobs in the list of jobs to include in the message body.

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

**Review by spikat — APPROVED** [review]

File: tasks/libs/pipeline_data.py [hunk 1]
```diff
@@ -130,3 +130,19 @@ def truncate_job_name(job_name, max_char_per_job=48):
...
     truncated_job_name = truncated_job_name[:max_char_per_job]
     return truncated_job_name
+
+
+# those jobs are `allow_failure: true` but still needs to be included
+# in fail reports
+jobs_allowed_to_fails_but_need_report = [re.compile(r'kitchen_test_security_agent.*')]
+
+
+def should_include_failed_job(job_name, allow_failure):
```
Comment (chouetz) [inline tasks/libs/pipeline_data.py:140]:
```suggestion
def should_report_job(job_name, allow_failure):
```
just a suggestion, sounds more explicit to me this way

File: tasks/libs/pipeline_data.py [hunk 2]
```diff
@@ -130,3 +130,19 @@ def truncate_job_name(job_name, max_char_per_job=48):
...
+    if not allow_failure:
+        return True
+
+    for job_to_include in jobs_allowed_to_fails_but_need_report:
+        if job_to_include.fullmatch(job_name):
+            return True
+
+    return False
```
Comment (chouetz) [inline tasks/libs/pipeline_data.py:148]:
```suggestion
    return any(pattern.fullmatch(job_name) for pattern in jobs_allowed_to_fails_but_need_report)
```
this is a more pythonic way for line 144-148

File: tasks/libs/pipeline_data.py [hunk 3]
```diff
@@ -130,3 +130,19 @@ def truncate_job_name(job_name, max_char_per_job=48):
     # We also want to avoid it being too long
     truncated_job_name = truncated_job_name[:max_char_per_job]
     return truncated_job_name
+
+
+# those jobs are `allow_failure: true` but still needs to be included
+# in fail reports
+jobs_allowed_to_fails_but_need_report = [re.compile(r'kitchen_test_security_agent.*')]
```
Comment (chouetz) [inline tasks/libs/pipeline_data.py:137]: As this is used only on your method below I would tend not to make it global, wdyt?

**Review by chouetz — APPROVED** [review]

File: tasks/libs/pipeline_data.py (hunk 1 above)
Comment (paulcacheux) [reply tasks/libs/pipeline_data.py:140]: I can do this yes

File: tasks/libs/pipeline_data.py (hunk 3 above)
Comment (paulcacheux) [reply tasks/libs/pipeline_data.py:137]: I was trying to match what was done with `infra_failure_logs` in the same file

**Pull request merged**